tls/s2n_config.h

```diff
@@ -90,6 +90,7 @@ struct s2n_config {
     s2n_cert_tiebreak_callback cert_tiebreak_cb;
 
     uint8_t mfl_code;
+    uint8_t cert_req_legacy_dss_enabled;
```

Comment:
Is there a reason this is a `uint8_t` and not a flag with the ones at the top of the struct definition?

Also, please add some documentation to the field indicating what it's used for. I know a lot of the other fields aren't documented, but something like this especially needs to be clear.

Reply:
In @lrstewart 's PR, she ran into the same thing and fixed it here: https://github.com/awslabs/s2n/pull/2359/files#diff-1fded1e92cdd5ab7f968a9154528b59efbd04a25de8f004cd431abb10477630d